### PR TITLE
WIP - UniDriver. Protractor uni driver factory creator + adapter.

### PR DESCRIPTION
**Pull request by mykas**
This is initial protractor driver factory creator for reuse in testkits. Adapter should not probably be here, but this is temporal adapter implementation until guys from unidriver approve this adapter pullrequest.

### CODE REVIEW

File: packages/wix-ui-test-utils/src/protractor/protractor.ts [hunk 1]
```diff
@@ -1,6 +1,23 @@
 import {$, ElementFinder} from 'protractor';
+import {protractorUniDriver} from './protractor-adapter';
+import {UniDriver} from 'unidriver';
```
Comment (lbelinsk) [inline packages/wix-ui-test-utils/src/protractor/protractor.ts]: You need to add unidriver to the package.json

File: packages/wix-ui-test-utils/src/protractor/protractor.ts [hunk 2]
```diff
@@ -1,6 +1,23 @@
...
+      ? driverFactory(obj.wrapper.$(`[data-hook='${obj.dataHook}']`))
+      : driverFactory($(`[data-hook='${obj.dataHook}']`));
+}
+
+export function protractorUniTestkitFactoryCreator<T>(
+  driverFactory: (base: UniDriver) => T
+) {
+  return (obj: { dataHook: string }) => {
+    const base = protractorUniDriver(
+      async () => await $(`[data-hook='${obj.dataHook}']`)
```
Comment (lbelinsk) [inline packages/wix-ui-test-utils/src/protractor/protractor.ts]: I don't think this should be under async await syntax. This is a lazy element selector getter and we expect the dev who uses this to wrap it with async await, not the getter itself

**Comment by lbelinsk** [comment]
@mykas looks great. I will ask Boris to review the pr in unidriver.

File: packages/wix-ui-test-utils/src/protractor/protractor.ts (hunk 2 above)
Comment (mykas) [reply packages/wix-ui-test-utils/src/protractor/protractor.ts]: But TS complains..

Comment (lbelinsk) [reply packages/wix-ui-test-utils/src/protractor/protractor.ts]: ok, so leave it.

File: packages/wix-ui-test-utils/src/protractor/protractor-adapter.ts [hunk 3]
```diff
@@ -0,0 +1,122 @@
+import {Locator, UniDriverList, UniDriver, MapFn} from 'unidriver';
```
Comment (hepiyellow) [inline packages/wix-ui-test-utils/src/protractor/protractor-adapter.ts]: @mykas Is this a copy of https://github.com/wix-incubator/unidriver/blob/master/src/lib/protractor/index.ts?
Why?
Please add comments if there is a good reason.

File: packages/wix-ui-test-utils/src/protractor/utils.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  const val = await fn();
+  if (!val) {
+    const now = Date.now();
+    await delay(retryDelay);
+    const delta = Date.now() - now;
+    return waitFor(fn, timeout - delta, retryDelay);
+  }
+};
+
+export const eventually = async (
```
Comment (hepiyellow) [inline packages/wix-ui-test-utils/src/protractor/utils.ts]: - Why don't you use wix-eventually ? (https://github.com/wix/wix-eventually/blob/master/index.js)
- Please document what this function does.

File: packages/wix-ui-test-utils/src/protractor/utils.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
+const delay = (ms: number) => new Promise(res => setTimeout(res, ms));
+
+export const waitFor = async (
```
Comment (hepiyellow) [inline packages/wix-ui-test-utils/src/protractor/utils.ts]: Missing documentation.

**Pull request merged**